platform/android/MapboxGLAndroidSDKTestApp/src/main/java/com/mapbox/mapboxsdk/testapp/activity/style/SymbolLayerActivity.java

```diff
@@ -108,6 +108,8 @@
   private MapboxMap mapboxMap;
   private MapView mapView;
 
+  private boolean iconAdded;
```

Comment:
Is this flag required? Shouldn't core stop requesting the missing icon if it's been already provided during the first callback?

Reply:
normally not but since our `Style#addImage` is asynchronous to optimise core input, we are  receiving multiple callbacks for missing images. I'm going to benchmark if we really need to keep that async, additionally we could wait with the callback invoke exposed by core.

Reply:
Ah yes, forgot about the async bitmap generation.
> additionally we could wait with the callback invoke exposed by core

I'm afraid this might add a couple of unnecessary render loops without the image. Would be good to benchmark as well.